R/utilities-quantity.R

```diff
@@ -160,7 +159,13 @@ setQuantityValuesByPath <- function(quantityPaths, values, simulation, units = N
       if (dimension == "") {
         next
       }
-      value <- toBaseUnit(quantityOrDimension = dimension, values = value, unit = units[[i]])
+      #If the unit is NULL, the value is assumend to be in base unit and no conversion
+      #in necessary
+      if (!is.null(units[[i]])){
+        mw <- simulation$molWeightFor(path)
```

Comment:
Good thing the method requires a path, so we can safely use it with the performance-optimized function.